### PR TITLE
tests: adding test cases for IWDG on disco stm32f3 board

### PR DESCRIPTION
**Pull request by FRASTM**
This patch enables the IWDG so that the testcase **tests/drivers/watchdog/wdt_basic_api**
can run on this **stm32f3_disco** board from STMicroelectronics

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

**Review by ABOSTM — APPROVED** [review]
LGTM

**Review by erwango — CHANGES_REQUESTED** [review]
In order to be run by sanitycheck script, you need to add '  - watchdog' in stm32f3_disco yaml supported section

**Comment by FRASTM** [comment]
new commit added with modified stm32f3_disco.yaml

**Review by erwango — CHANGES_REQUESTED** [review]
Having 2 commits in this case seems a bit overkill, please squash.
Also, please update the board documentation.

**Comment by FRASTM** [comment]
now squashed in a single commit

File: boards/arm/stm32f3_disco/doc/index.rst [hunk 1]
```diff
@@ -65,6 +65,7 @@ STM32F3DISCOVERY Discovery kit provides the following hardware components:
 - USB 2.0 full speed interface
 - Infrared transmitter
 - DMA Controller
+- independent watchdog
```
Comment (erwango) [inline boards/arm/stm32f3_disco/doc/index.rst:68]: I was rather referring to the "Supported features" section. Please update

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/arm/stm32f3_disco/doc/index.rst (hunk 1 above)
Comment (FRASTM) [reply boards/arm/stm32f3_disco/doc/index.rst]: ok, moved

**Review by erwango — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**